### ISSUE TITLE
[Suggestion] Notify players when a bounty has been completed.

### ISSUE THREAD

**Issue by katubug** [ejektaflex/Bountiful#38]
One of my players suggested a visual change to the item icon, such as a stamp or a checkmark appearing in the corner. Or a little text notification. Something that lets players know when they've completed a bounty's objective without having to hover over it.

**Comment by ejektaflex** [comment]
That's a good idea. I'll keep it up here as a suggestion, I'll look into implementing it.

One thing to keep in mind, is that in order to notify players when a bounty has been completed it has to check on a regular basis for each bounty, which uses up server tick time. As it stands now, it only checks whether the bounty has been completed when you try to turn it in. Notifying the player would require checking every bounty in your inventory for completion every second, which has some overhead cost for the server.

Granted, that's pretty minimal compared to how some mods work. I think that this would make a good optional feature, which could be turned off.

**Comment by ejektaflex** [comment]
Okay @katubug , so I have redesigned the entire tooltip for Bountiful v3.0 for when it releases for Minecraft v1.14. The tooltip should more clearly indicate how close you are to finishing each objective. As for a visual indicator, I plan on adding an optional (default off) durability bar for bounties with two different modes:
* A) Durability indicates time left to complete the bounty. It shrinks as you get closer to running out of time.
* B) Durability indicates how close you are to completing the bounty. A full durability bar means that you've completed all of it's objectives.

Hopefully B will be a good solution for you, once I add it!

**Comment by katubug** [comment]
Hi, thanks for the update! That all sounds fantastic. Thank you so much for
all the hard work you've put into this mod. 💗

**Comment by ejektaflex** [comment]
Alright, I've tried to implement this (the damage on the item) but it seems quite difficult to do, since now when items are in the inventory they seemingly don't have any knowledge of the player whose inventory they're in, something needed to calculate how far that player is to completing their bounty. As such, I've already completely changed the tooltips for v3.0, so I hope that's a good enough way of notifying players when they've finished the bounty (It seems to be much clearer and incentivize players to finish them on my server!). Closing this for now, but keeping it as an idea for later if I try to pursue it again.
